marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -51,7 +54,11 @@ function addPropertiesToSearchResponse(entityName, searchResponse, propertiesToD
 
     let instance = null;
     try {
-      instance = cts.doc(result.uri).toObject().envelope.instance;
+      const instanceArray = getEntityInstance(result.uri);
+      instance = instanceArray.find(instance => Object.keys(instance)[0] === entityName &&
```

Comment:
I'm hoping this code won't be necessary if you call fn.head on the result from es.instanceJsonFromDocument.